optimade/validator/validator.py

```diff
@@ -169,24 +177,46 @@ def wrapper(validator, *args, **kwargs):
             request = validator.client.last_request
         except AttributeError:
             request = validator.base_url
+
         if result is not None:
-            validator.success_count += 1
+            if not optional:
+                validator.success_count += 1
+            else:
+                validator.optional_success_count += 1
+            message = f"✔: {request} - {msg}"
             if validator.verbosity > 0:
-                print_success(f"✔: {request} - {msg}")
+                if optional:
+                    print(message)
```

Comment:
Why can we not have success printing here as well? As well as the other places.

Reply:
The idea was that the optional tests aren't highlighted, otherwise the screen gets full with green ticks (hopefully) and the "optional tests" heading disappears off the screen.

Reply:
Okay. I still think it would be nice with the pretty formatting, but I see your point. I guess the output should be re-done a bit for verbosity levels 0 and 1, or perhaps just 1, before we can comfortably then include both pretty formatting and a good overview?

Reply:
What would you suggest? Currently it's just a two stage process: mandatory tests printed with colour, then optional tests printed without colour. Then a final summary of both.

Reply:
Yup. But the messages are very verbose, I guess this is why you didn't want the pretty print, since there will be a lot of green characters and one might miss the difference between the two sections?
So I was merely thinking about making the text for each test more minimalistic for verbosity 1 (including the pretty printing), keeping verbosity 0 like it already is, and then have the current verbosity 1 for verbosity 2 instead? And then possibly first include debug logs at verbosity 3+?

Reply:
In the end, it's fine as it is now, since the "new" verbosity 1 output I mention will take some time to implement as well - and isn't crucial, but rather an enhancement for a later PR, I think.

Reply:
I get you, but how could you summarise it more? At the moment it just prints tick/cross and the request and a message about serialization. Just tick/cross and the request, excluding the base url?

Reply:
Also happy to leave it for now though :)

Reply:
Hmm, would maybe do something different completely - similar to what pytest does? Like write `Filter tests` followed by dots (.) if successful and x'es if not, or an F or something?
This is why I think we should just leave it as you have it now and then consider it an enhancement for later endeavours, since it's not at all clear how it should be done at this point, and it may also involve some serious coding... or not, I don't know 😅